internal/app/machined/pkg/controllers/network/link_config.go

```diff
@@ -276,58 +282,78 @@ func (ctrl *LinkConfigController) parseMachineConfiguration(logger *zap.Logger,
 		}
 
 		for _, linkName := range device.Bond().Interfaces() {
-			if _, exists := bondedLinks[linkName]; exists {
+			if bondName, exists := bondedLinks[linkName]; exists && bondName != device.Interface() {
 				logger.Sugar().Warnf("link %q is included into more than two bonds", linkName)
 			}
 
 			bondedLinks[linkName] = device.Interface()
 		}
 	}
 
+	linkMap := map[string]*network.LinkSpecSpec{}
```

Comment:
correctly handle multiple entries for the same interface in the machine configuration (our machine configuration format is weird!)